airavata-api/src/main/java/org/apache/airavata/api/server/handler/AiravataServerHandler.java

```diff
@@ -2079,15 +1602,16 @@ public boolean validateExperiment(AuthzToken authzToken, String airavataExperime
     @Override
     @SecurityCheck
     public ExperimentStatus getExperimentStatus(AuthzToken authzToken, String airavataExperimentId) throws TException {
-        RegistryService.Client regClient = registryClientPool.getResource();
         try {
-            ExperimentStatus result = regClient.getExperimentStatus(airavataExperimentId);
-            registryClientPool.returnResource(regClient);
+            ExperimentStatus result = airavataService.getExperimentStatus(airavataExperimentId);
             return result;
+        } catch (org.apache.airavata.registry.cpi.RegistryException e) {
+            AiravataSystemException exception = new AiravataSystemException();
+            exception.setMessage(e.getMessage());
+            throw exception;
```

Comment:
This catch block doesn't give any advantage. Better to remove and let the 'catch (Exception e)' block handles it. And make sure to log the errors